src/javascript/app/Modules/SmartChart/Containers/smart-chart.jsx

```diff
@@ -38,6 +38,15 @@ class Chart extends React.Component {
     );
 
     render() {
+        const granularity = this.props.granularity;
+        const duration    = this.props.end_epoch - this.props.start_epoch;
+
+        const margin = granularity === 0 ? (Math.max(300, (30 * duration) / (60 * 60) || 0)) : 3 * granularity;
+
+        // load more ticks so we can draw one tick before and one tick after contract
+        const start_margin = ((this.props.start_epoch) - margin).toFixed(0);
+        const end_margin   = (this.props.end_epoch + margin).toFixed(0);
```

Comment:
@negarn 'toFixed` returns String, you use bitwise OR to remove the decimal points.
```suggestion
        const end_margin   = (this.props.end_epoch + margin) | 0;
```